code/modules/mob/living/simple_animal/eldritch_demons.dm

```diff
@@ -0,0 +1,267 @@
+	if(health == maxHealth)
+		if(back)
+			back.heal()
+			return
+		else
+			current_stacks++
+			if(current_stacks >= stacks_to_grow)
+				var/mob/living/simple_animal/hostile/eldritch/armsy/prev = new type(drop_location(),spawn_more = FALSE)
+				icon_state = "armsy_mid"
+				icon_living =  "armsy_mid"
+				back = prev
+				prev.icon_state = "armsy_end"
+				prev.icon_living = "armsy_end"
+				prev.front = src
+				prev.AIStatus = AI_OFF
+
+				current_stacks = 0
+
+	adjustBruteLoss(-maxHealth/2)
+	adjustFireLoss(-maxHealth/2)
+	adjustToxLoss(-maxHealth/2)
+	adjustOxyLoss(-maxHealth/2)
+
+/mob/living/simple_animal/hostile/eldritch/armsy/AttackingTarget()
+	if(istype(target,/obj/item/bodypart/r_arm) || istype(target,/obj/item/bodypart/l_arm))
+		target.Destroy()
+		heal()
+		return
+	if(back)
+		back.target = target
+		back.AttackingTarget()
+	if(!Adjacent(target))
+		return
+	do_attack_animation(target)
+	if(iscarbon(target))
+		var/mob/living/carbon/C = target
+		if(HAS_TRAIT(C, TRAIT_NODISMEMBER))
+			return
+		var/list/parts = list()
+		for(var/X in C.bodyparts)
+			var/obj/item/bodypart/BP = X
+			if(BP.body_part != HEAD && BP.body_part != CHEST && BP.body_part != LEG_LEFT && BP.body_part != LEG_RIGHT)
+				if(BP.dismemberable)
+					parts += BP
+		if(prob(10))
+			var/obj/item/bodypart/BP = pick(parts)
+			if(BP)
+				BP.dismember()
+
+	. = ..()
+
+/mob/living/simple_animal/hostile/eldritch/armsy/prime
+	name = "Lord of the Night"
+	real_name = "Master of Decay"
+	maxHealth = 400
+	health = 400
+	melee_damage_lower = 20
+	melee_damage_upper = 25
+
+/mob/living/simple_animal/hostile/eldritch/armsy/prime/New(spawn_more, len)
+	var/matrix/M = matrix()
+	M.Scale(1.4,1.4)
+	src.transform = M
+	. = ..()
+
+/mob/living/simple_animal/hostile/eldritch/rust_spirit
+	name = "Rust Walker"
+	real_name = "Rusty"
+	desc = "Incomprehensible abomination actively seeping life out of it's surrounding."
+	icon_state = "rust_walker"
+	status_flags = CANPUSH
+	icon_living = "rust_walker"
+	maxHealth = 50
+	health = 50
+	melee_damage_lower = 15
+	melee_damage_upper = 20
+	sight = SEE_TURFS
+
+/mob/living/simple_animal/hostile/eldritch/rust_spirit/Initialize()
+	. = ..()
+	AddSpell(/obj/effect/proc_holder/spell/aoe_turf/rust_conversion/small)
+	AddSpell(/obj/effect/proc_holder/spell/targeted/projectile/dumbfire/rust_wave/short)
+
+/mob/living/simple_animal/hostile/eldritch/ash_spirit
+	name = "Ash Man"
+	real_name = "Ashy"
+	desc = "Incomprehensible abomination actively seeping life out of it's surrounding."
+	icon_state = "ash_walker"
+	status_flags = CANPUSH
+	icon_living = "ash_walker"
+	maxHealth = 50
+	health = 50
+	melee_damage_lower = 15
+	melee_damage_upper = 20
+	sight = SEE_TURFS
+
+/mob/living/simple_animal/hostile/eldritch/ash_spirit/Initialize()
+	. = ..()
+	AddSpell(/obj/effect/proc_holder/spell/targeted/ethereal_jaunt/shift/ash)
+	AddSpell(/obj/effect/proc_holder/spell/pointed/ash_cleave/long)
```

Comment:
give these guys, like, the ability to heal people or something

and/or more health

and/or that morph ability to copy the appearances of people and objects, so that they can act as doppelgangers of you (and/or your victims) or something

Reply:
and/or that knock spell that flesh stalkers get